src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -296,9 +296,12 @@ export class SavedObjectsService
     legacyTypes.forEach(type => this.typeRegistry.registerType(type));
     this.validations = setupDeps.legacyPlugins.uiExports.savedObjectValidations || {};
 
-    const importableExportableTypes = getImportableAndExportableTypes(
-      setupDeps.legacyPlugins.uiExports
-    );
+    const getImportableAndExportableTypes = () => {
+      return this.typeRegistry
+        .getAllTypes()
+        .map(type => type.name)
+        .filter(type => this.typeRegistry.isImportableAndExportable(type));
+    };
```

Comment:
Same 'issue' than with the migrator, The (NP registered) types are not registered until the end of the setup phase, I now need an accessor to retrieve them instead of raw values.

Reply:
Couldn't these routes use the `context.savedObjects.typeRegistry` to get this data? If so, we could just add a method to the type registry that does this filtering.

Reply:
Good catch. I actually did that before adding the typeRegistry to the context. The routes should use the context instead.

Reply:
Hum, actually, the types are needed outside of the handler:

```ts
  const typeSchema = schema.string({
    validate: (type: string) => {
      if (!getSupportedTypes().includes(type)) {
        return `${type} is not exportable`;
      }
    },
  });
```

Guess I can move the validation in the handler's body instead.